ethcore/src/engines/clique/mod.rs

```diff
@@ -0,0 +1,766 @@
+			return Ok(state.clone());
+		}
+		// If we are looking for an checkpoint block state, we can directly reconstruct it.
+		if header.number() % self.epoch_length == 0 {
+			let state = self.new_checkpoint_state(header)?;
+			block_state_by_hash.insert(header.hash(), state.clone());
+			return Ok(state);
+		}
+		// BlockState is not found in memory, which means we need to reconstruct state from last checkpoint.
+		match self.client.read().as_ref().and_then(|w| w.upgrade()) {
+			None => {
+				return Err(EngineError::RequiresClient)?;
+			}
+			Some(c) => {
+				let last_checkpoint_number = header.number() - header.number() % self.epoch_length as u64;
+				debug_assert_ne!(last_checkpoint_number, header.number());
+
+				let mut chain: &mut VecDeque<Header> = &mut VecDeque::with_capacity(
+					(header.number() - last_checkpoint_number + 1) as usize);
+
+				// Put ourselves in.
+				chain.push_front(header.clone());
+
+				// populate chain to last checkpoint
+				loop {
+					let (last_parent_hash, last_num) = {
+						let l = chain.front().expect("chain has at least one element; qed");
+						(*l.parent_hash(), l.number())
+					};
+
+					if last_num == last_checkpoint_number + 1 {
+						break;
+					}
+					match c.block_header(BlockId::Hash(last_parent_hash)) {
+						None => {
+							return Err(BlockError::UnknownParent(last_parent_hash))?;
+						}
+						Some(next) => {
+							chain.push_front(next.decode()?);
+						}
+					}
+				}
+
+				// Catching up state, note that we don't really store block state for intermediary blocks,
+				// for speed.
+				let backfill_start = time::Instant::now();
+				trace!(target: "engine",
+						"Back-filling block state. last_checkpoint_number: {}, target: {}({}).",
+						last_checkpoint_number, header.number(), header.hash());
+
+				// Get the state for last checkpoint.
+				let last_checkpoint_hash = *chain.front()
+					.expect("chain has at least one element; qed")
+					.parent_hash();
+
+				let last_checkpoint_header = match c.block_header(BlockId::Hash(last_checkpoint_hash)) {
+					None => return Err(EngineError::CliqueMissingCheckpoint(last_checkpoint_hash))?,
+					Some(header) => header.decode()?,
+				};
+
+				let last_checkpoint_state = match block_state_by_hash.get_mut(&last_checkpoint_hash) {
+					Some(state) => state.clone(),
+					None => self.new_checkpoint_state(&last_checkpoint_header)?,
+				};
+
+				block_state_by_hash.insert(last_checkpoint_header.hash(), last_checkpoint_state.clone());
+
+				// Backfill!
+				let mut new_state = last_checkpoint_state.clone();
+				for item in chain {
+					new_state.apply(item, false)?;
+				}
+				new_state.calc_next_timestamp(header.timestamp(), self.period)?;
+				block_state_by_hash.insert(header.hash(), new_state.clone());
+
+				let elapsed = backfill_start.elapsed();
+				trace!(target: "engine",
+						"Back-filling succeed, took {} ms.",
+						// replace with Duration::as_millis after rust 1.33
```

Comment:
I thought we'd resolved this, or am I mistaken?

Reply:
Never fixed but I don't think that the trivial thingy is worth the effort of requiring `Rust 1.33` to compile there are still some platforms (no rustup) that can't be updated for every release

Reply:
I can fix it if you don't agree!

Reply:
No, it's fine we can leave it as is. I think we should at least track it with an issue though so it doesn't get forgotten in the land of immortal TODOs